File: frontend/src/constants/recordStatus.js
// Trạng thái bản ghi theo đặc tả URD (vòng đời Maker-Checker)
// Mã trạng thái nghiệp vụ của bản ghi Client (Customer.RecordStatus)
export const RECORD_STATUS = {
  PENDING_INSERT: 'PI',
  PENDING_UPDATE: 'PU',
  PENDING_DELETE: 'PD',
  ACTIVE: 'A',
  REJECTED: 'R',
  DELETED: 'D'
}

// Nhãn hiển thị VN/EN cho từng trạng thái
export const RECORD_STATUS_LABELS = {
  [RECORD_STATUS.PENDING_INSERT]: { vi: 'Chờ duyệt thêm', en: 'Pending Insert' },
  [RECORD_STATUS.PENDING_UPDATE]: { vi: 'Chờ duyệt sửa', en: 'Pending Update' },
  [RECORD_STATUS.PENDING_DELETE]: { vi: 'Chờ duyệt xóa', en: 'Pending Delete' },
  [RECORD_STATUS.ACTIVE]: { vi: 'Đã duyệt', en: 'Active' },
  [RECORD_STATUS.REJECTED]: { vi: 'Từ chối', en: 'Rejected' },
  [RECORD_STATUS.DELETED]: { vi: 'Đã xóa', en: 'Deleted' }
}

// Class CSS badge tương ứng để tô màu trạng thái trên grid
export const RECORD_STATUS_BADGE = {
  [RECORD_STATUS.PENDING_INSERT]: 'badge-warning',
  [RECORD_STATUS.PENDING_UPDATE]: 'badge-warning',
  [RECORD_STATUS.PENDING_DELETE]: 'badge-warning',
  [RECORD_STATUS.ACTIVE]: 'badge-success',
  [RECORD_STATUS.REJECTED]: 'badge-danger',
  [RECORD_STATUS.DELETED]: 'badge-default'
}

// Lấy nhãn hiển thị theo ngôn ngữ (mặc định tiếng Việt)
export function recordStatusLabel(code, lang = 'vi') {
  const entry = RECORD_STATUS_LABELS[code]
  if (!entry) return code || '—'
  return entry[lang] || entry.vi
}

// Lấy class badge tương ứng trạng thái
export function recordStatusBadge(code) {
  return RECORD_STATUS_BADGE[code] || 'badge-default'
}

File: The_Last_Dance_Project/Constants/RecordStatus.cs
namespace The_Last_Dance_Project.Constants
{
    /// <summary>
    /// Trạng thái giao dịch Maker-Checker lưu tại MTTRAN.MtlStatus.
    /// </summary>
    public static class TransactionStatus
    {
        public const string Pending = "N";    // Chờ duyệt
        public const string Approved = "A";   // Đã duyệt
        public const string Rejected = "R";   // Bị từ chối
        public const string Cancelled = "C";  // Maker hủy
    }

    /// <summary>
    /// Loại hành động của giao dịch Maker-Checker lưu tại MTTRAN.MtlType.
    /// Lưu ý: cột MTLTYPE giới hạn 3 ký tự nên dùng mã ngắn I/U/D.
    /// </summary>
    public static class TransactionType
    {
        public const string Insert = "I";
        public const string Update = "U";
        public const string Delete = "D";

        /// <summary>Chuẩn hóa action đầu vào (CREATE/ADD/NEW -> I, ...).</summary>
        public static string Normalize(string? action)
        {
            var a = (action ?? string.Empty).Trim().ToUpperInvariant();
            return a switch
            {
                "CREATE" or "ADD" or "NEW" or "INSERT" or "I" => Insert,
                "UPDATE" or "EDIT" or "MODIFY" or "U" => Update,
                "DELETE" or "REMOVE" or "DEL" or "D" => Delete,
                _ => a.Length > 3 ? a.Substring(0, 3) : a
            };
        }
    }

    /// <summary>
    /// Trạng thái bản ghi nghiệp vụ (Client) theo đặc tả URD.
    /// Dùng cho Customer.RecordStatus và hiển thị trên giao diện.
    /// </summary>
    public static class RecordStatus
    {
        public const string PendingInsert = "PI"; // Chờ duyệt thêm
        public const string PendingUpdate = "PU"; // Chờ duyệt sửa
        public const string PendingDelete = "PD"; // Chờ duyệt xóa
        public const string Active = "A";         // Đã duyệt
        public const string Rejected = "R";       // Từ chối
        public const string Deleted = "D";        // Đã xóa
    }
}
